base/doc/usrguide.tex

```diff
@@ -794,6 +794,59 @@ \subsection{Commands at the start of tabular cells}
 \end{tabular}
 \end{verbatim}
 
+\subsection{Using the verbatim argument types}
+
+As described above, the \texttt{v}-type argument may be viewed as similar to
+\cs{verb}. Before looking at exactly what that means, it is important to
+highlight some key differences. Most notably, \emph{grabbing} a verbatim-like
+argument is separate from \emph{typesetting} it: the latter is covered in the
+next section.
+
+When grabbing a \texttt{v}-type argument, \LaTeX{} first uses the kernel
+command \cs{dospecials} to turn off the \enquote{special} nature of characters.
+It then makes both spaces and tabs \enquote{active}, so they can be given a
+custom definition. Any other characters are grabbed as-is: this means that if
+any characters have been made \enquote{special} and are not listed in
+\cs{dospecials}, an error will arise (see below).
+
+Characters are grabbed two identical delimiters: in contrast to \cs{verb}, the
+characters \texttt{\textbackslash}, |{|, |}| and |%| \emph{cannot be used}. If
+any of the grabbed tokens have \enquote{special} meaning, an error will be
+issued.
+
+For the \texttt{+v}-type argument, which allows line breaks in input,
+newline characters are converted into \cs{obeyedline} commands. This means that
+the grabbed tokens can be used directly in typesetting, while a local
+redefinition of \cs{obeyedline} can be used to achieve other outputs. For
+example, to retain blank lines whilst typesetting, one could use
+\begin{verbatim}
+\renewcommand*\obeyedline{\mbox{}\par}
+\end{verbatim}
+along with other setting that may be appropriate (e.g., using a monospaced
+font).
+
+Some additional details that may be useful for those with more \TeX{}
+knowledge: do not worry if this does not make sense to you! Grabbed tokens are
+all stored as either active or \enquote{other}: spaces, tabs and non-ASCII
+codepoints in $8$-bit engines are active, everything else is detokenized. As
+such, token-based comparisons are likely to fail unless set up properly.
+
+\subsection{Typesetting verbatim-like material}
+
+In contrast to \cs{verb}, the \texttt{v}-type argument is only about
+\emph{grabbing} the argument, not \emph{typesetting} it. As such, features that
+users often associate with \enquote{verbatim} are not automatically activated,
+e.g., a monospaced font. Material grabbed by the \texttt{v}-type argument does
+not automatically suppress ligatures: with modern \TeX{} engines, this largely
+can be done without the token manipulation which \cs{verb} uses. (In \cs{verb},
+ligatures are suppressed by making characters active and inserting a zero-width
+kern before the character itself.)
+
+The \cs{verb} command also selects a monotype font: this is not intrinsic
+to verbatim material, so will need to be set up using for example \cs{ttfamily}.
+Similarly, the \texttt{verbatim} environment sets up the meaning of \cs{par}
+suitable for breaking lines.
```

Comment:
Maybe need to add:
. . . , so this may also need to be set up (see example above) ??